### ISSUE TITLE
Rule Request: Quick Discouraged Pending Test

### ISSUE THREAD

**Issue by ornithocoder** [realm/SwiftLint#1909]
There are hella reasons to mark a test as pending. But not many reasons to keep tests as pending forever in the code base. Tests that aren't running are dead code; they make sense as a temporary solution, but not for the eternity. Pending tests should be monitored, reviewed, and refactored every now and then.

This is a rule request to introduce a new opt-in rule, `quick_discouraged_pending_test`, to warn against pending tests. While some teams might want to have this always on (or off), others might prefer to run it sporadically.

## Triggering Examples

```swift
class TotoTests: QuickSpec {
   override func spec() {
       ↓xdescribe("foo") { }
   }
}

```

```swift
class TotoTests: QuickSpec {
   override func spec() {
       ↓xcontext("foo") { }
   }
}

```

```swift
class TotoTests: QuickSpec {
   override func spec() {
       ↓xit("foo") { }
   }
}

```

```swift
class TotoTests: QuickSpec {
   override func spec() {
       describe("foo") {
           ↓xit("bar") { }
       }
   }
}

```

```swift
class TotoTests: QuickSpec {
   override func spec() {
       context("foo") {
           ↓xit("bar") { }
       }
   }
}

```

```swift
class TotoTests: QuickSpec {
   override func spec() {
       describe("foo") {
           context("bar") {
               ↓xit("toto") { }
           }
       }
   }
}

```

```swift
class TotoTests: QuickSpec {
   override func spec() {
       ↓pending("foo")
   }
}

```

```swift
class TotoTests: QuickSpec {
   override func spec() {
       ↓xitBehavesLike("foo")
   }
}

```

**Comment by ornithocoder** [comment]
Merged in #1911.
